lib/pipeline.mli

```diff
@@ -0,0 +1,34 @@
+(** Common logic for building a CI pipeline. *)
+
+val experimental_variant : string -> bool
+(** Check whether a variant is considered experimental.
+
+    If it is experimental we allow those builds to fail without failing the
+    overall build for a commit. *)
+
+val summarise :
+  (string
+  * (([< `Built | `Checked ], [< `Active of 'a | `Msg of string ]) result * 'b))
+  list ->
+  (unit, [> `Active of [> `Running ] | `Msg of string ]) result
```

Comment:
I think it's the usual appeal of not having to import the common module it is defined everywhere we want to use it. It could be improved if someone wanted to have a go.

Reply:
Sure, maybe in another PR then :+1: